test/framework/integration_framework/integration_test_framework.cpp

```diff
@@ -265,7 +268,11 @@ namespace integration_framework {
               statuses.push_back(*std::static_pointer_cast<
                                  shared_model::proto::TransactionResponse>(s));
             },
-            [&bar] { bar.wait(); });
+            [&cv, &m, &processed] {
+              std::lock_guard<std::mutex> lock(m);
```

Comment:
lock seems redundant here, what is its purpose? `processed` cannot be modified concurrently because on_complete is called once only.

Reply:
According to https://en.cppreference.com/w/cpp/thread/condition_variable it is needed